iotedgehubdev/edgecert.py

```diff
@@ -37,8 +40,62 @@ def generate_self_signed_certs(self):
         prefixes = [EdgeConstants.EDGE_AGENT_CA, EdgeConstants.EDGE_DEVICE_CA]
         cert_util.chain_ca_certs(EdgeConstants.EDGE_CHAIN_CA, prefixes, self.certs_dir)
 
+    # Generate IoT Edge device CA to be configured in IoT Edge runtime
+    def generate_device_ca(self, valid_days, overwrite_existing, trusted_ca, trusted_ca_key, trusted_ca_key_passphase):
+        # Function level variables
+        root_ca_cert_path = os.path.join(self.certs_dir, EdgeConstants.ROOT_CA_ID + EdgeConstants.CERT_SUFFIX)
+        root_ca_key_path = os.path.join(self.certs_dir, EdgeConstants.ROOT_CA_ID + EdgeConstants.KEY_SUFFIX)
+        device_ca_cert_path = os.path.join(self.certs_dir, EdgeConstants.DEVICE_CA_ID + EdgeConstants.CERT_SUFFIX)
+        device_ca_key_path = os.path.join(self.certs_dir, EdgeConstants.DEVICE_CA_ID + EdgeConstants.KEY_SUFFIX)
+        device_ca_cert_chain_path = os.path.join(self.certs_dir, EdgeConstants.DEVICE_CA_ID + EdgeConstants.FULLCHAIN_CERT_SUFFIX)
+        create_root_ca = not (trusted_ca and trusted_ca_key)
+        # Check existing cert files
+        existing_files = self.__get_existing_files__(device_ca_cert_path, device_ca_key_path, device_ca_cert_chain_path)
+        if create_root_ca:
+            existing_files.extend(self.__get_existing_files__(root_ca_cert_path, root_ca_key_path))
+
+        if len(existing_files) > 0:
```

Comment:
Move check existing/force stuff to cli? In this method it only cares to generate and output result to the folder. And don't care the force value